app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -6,11 +6,37 @@ def create_destination
   def determine_placement
     availability_zone = placement
     options[:dest_availability_zone] = [availability_zone.try(:id), availability_zone.try(:name)]
-    signal :prepare_provision
+    signal :prepare_volumes
+  end
+
+  def prepare_volumes
+    # volumes example: [{name: "vol_name", size: 1}]
+    if miq_request.options[:volumes]
+      phase_context[:requested_volumes] = create_requested_volumes(miq_request.options[:volumes])
+      signal :poll_volumes_complete
+    else
+      signal :prepare_provision
+    end
+  end
+
+  def poll_volumes_complete
+    status, status_message = do_volume_creation_check(phase_context[:requested_volumes])
+    status_message = "completed prepare provision work queued" if status
+    message = "Volume creation is #{status_message}"
+    _log.info(message)
+    update_and_notify_parent(:message => message)
+    if status
+      signal :prepare_provision
+    else
+      requeue_phase
+    end
   end
 
   def start_clone_task
     update_and_notify_parent(:message => "Starting Clone of #{clone_direction}")
+    if phase_context[:requested_volumes]
+      phase_context[:clone_options][:block_device_mapping_v2] = phase_context[:requested_volumes]
```

Comment:
this is still missing the image, otherwise you will boot from a volume (I see that you set bootindex 0 of the first)

:block_device_mapping_v2 => [{
                   :source_type           => "image",
                   :destination_type      => "local",
                   :boot_index            => 0,
                   :delete_on_termination => true,
                   :uuid                  => image.id
                 }, {

Given you are picking an image as a first step, I would say the image should be there. So I would say this will be your first choice in UI, where you will have source the image. The other options could be probably choosable, if you want to deploy it to local or volume.

You might want to discuss this with @Loicavenel